rfcs/0001-extended-lookup-tables.md

```diff
@@ -0,0 +1,503 @@
+contains multiple values -- where we want to ensure that the prover used an
+entry `(x, y, z)` from the table, and not some parts of different entries, or
+some other value entirely. In order to do this, we generate a randomising field
+element `joint_combiner`, where the randomness depends on the circuit witness,
+and then compute `x + y * joint_combiner + z * joint_combiner^2`. Notice that,
+if the prover wants to select some `(a, b, c)` not in the table and claim that
+it in fact was, they are not able to use the knowledge of the `joint_combiner`
+to choose their `(a, b, c)`, since their choice will change the value of the
+`joint_combiner`.
+
+This is a standard technique for checking multiple equalities at once by
+checking a single equality; to use it, we must simply ensure that the
+randomness depends on all of the (non-constant) values in the equality.
+
+We propose extending each table with an extra column, which we will call the
+`TableID` column, which associates a different number with each table. Since
+the sizes of all tables must be known at constraint system time, we can treat
+this as a fixed column. For example, this converts a lookup `(a, b, c)` in the
+table with ID `1` into a lookup of `(a, b, c, 1)` in the combined table, and
+ensures that any value `(x, y, z)` from table ID `2` will not match that
+lookup, since its representative is (x, y, z, 2)`.
+
+To avoid any interaction between the `TableID` column and any other data in
+the tables, we use `TableID * joint_combiner^max_joint_size` to mix in the
+`TableID`, where `max_joint_size` is the maximum length of any lookup or table
+value. This also simplifies the 'joining' of the `table_id` polynomial
+commitment, since we can straightforwardly scale it by the constant
+`joint_combiner^max_joint_size` to find its contribution to every row in the
+combined table.
+
+### Sampling the `joint_combiner`
+
+Currently, the `joint_combiner` is sampled using randomness that depends on the
+witness. However, where we have runtime or side-loaded tables, a malicious
+prover may be able to select values in those tables that absue knowledge of
+`joint_combiner` to create collisions.
+
+For example, to maliciously insert a value `(x, y, z)` for table with ID 1, the
+prover could select the value `x + y * joint_combiner + z * joint_combiner^2 +
+-1 * joint_combiner^max_joint_size` for an entry in table 2.
+
+Thus, we need to ensure that the `joint_combiner` depends on the values in
+runtime or side-loaded tables, so that the values in these tables cannot be
+chosen using knowledge of the `joint_combiner`. To do this, we must use the
+commitments to the values in these tables as part of the source of randomness
+for `joint_combiner` in the prover and verifier.
+
+In particular, this means that we must have a commitment per column for each
+type of table, so that the verifier can confirm the correct construction of the
+combined table.
+
+### Representing the combined fixed table
```

Comment:
I don't think the formulas are correct in this section. I think it'd be good to have diagram to match the explanation. I made this but I'm confused as to what happens after the padding step

![IMG_5275](https://user-images.githubusercontent.com/1316043/148098791-e477d9cd-d1d0-4510-9f67-9ba0964b4f45.jpg)